cmd/pack3d/main.go

```diff
@@ -104,19 +116,41 @@ func main() {
 	best := 1e9  //the best score
 	/* This loop is to find the best packing stl, thus it will generate mutiple output
 Add 'break' in the loop to stop program */
+	start := time.Now()
 	for {
 		model, ntime = model.Pack(annealingIterations, nil, singleStlSize, frameSize)
 		if ntime >= 19990{
-			model.Reset()
-			continue
+			if time.Since(start).Seconds() <= 20{
+				model.Reset()
+				continue
+			}else{
+				fmt.Println("Cannot get a result, please decrease your numbers of STL")
+				break
+			}
+
 		}
 		score := model.Energy()  // score < 1, the smaller the better
 		if score < best {
 			best = score
 			done = timed("writing mesh")
-			model.Mesh().SaveSTL(fmt.Sprintf("pack3d-%.3f.stl", score))  // calling the mesh function in model
+			transformation := model.Transformation()
+			for j:=0; j<len(transformation); j++{
+				t := transformation[j]
+				transMatrix := [4][4]float64{{t.X00, t.X01, t.X02, t.X03},{t.X10, t.X11, t.X12, t.X13},{t.X20, t.X21, t.X22, t.X23},{t.X30, t.X31, t.X32, t.X33}}
+				content := Data{filenames[j], transMatrix}
+				data = append(data, content)
+			}
+			b, err := json.Marshal(data)
```

Comment:
please also make `b` into a better variable name. 'positions_json' may be a good. name.